DSharpPlus.Commands/ContextChecks/DirectMessageUsageCheck.cs

```diff
@@ -0,0 +1,42 @@
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+using DSharpPlus.Commands.Trees;
+
+namespace DSharpPlus.Commands.ContextChecks;
+
+internal sealed class DirectMessageUsageCheck : IContextCheck<DirectMessageUsageAttribute>
+{
+    public ValueTask<string?> ExecuteCheckAsync(DirectMessageUsageAttribute attribute, CommandContext context)
+    {
+        if (context.Channel.IsPrivate && attribute.Usage is not DirectMessageUsage.DenyDMs)
+        {
+            return ValueTask.FromResult<string?>(null);
+        }
+        else if (!context.Channel.IsPrivate && attribute.Usage is not DirectMessageUsage.RequireDMs)
```

Comment:
Merge with previous if statement, removing `else` block

Reply:
how do you suggest that be accomplished without an 193 character long line, which is twice what'd be generally considered reasonable? - it doesn't do any harm here, anyway, and i personally think this is clearer

Reply:
To wrap the `&&` and `||` lines

Reply:
```cs
    public ValueTask<string?> ExecuteCheckAsync(DirectMessageUsageAttribute attribute, CommandContext context)
    {
        if ((context.Channel.IsPrivate && attribute.Usage is not DirectMessageUsage.DenyDMs)
        || (!context.Channel.IsPrivate && attribute.Usage is not DirectMessageUsage.RequireDMs))
        {
            return ValueTask.FromResult<string?>(null);
        }

        string requirement = attribute.Usage switch
        {
            DirectMessageUsage.DenyDMs => "denies DM usage",
            DirectMessageUsage.RequireDMs => "requires DM usage",
            _ => throw new UnreachableException
            (
                $"DirectMessageUsageCheck reached an unreachable branch: {attribute.Usage}, IsPrivate was {context.Channel.IsPrivate}"
            )
        };

        return ValueTask.FromResult<string?>
        (
            $"The executed command {requirement} but was executed {(context.Channel.IsPrivate ? "inside a DM" : "outside a DM")}."
        );
    }
```